### PR TITLE
enh: use table names in assistant builder

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

In assistant builder, we want to display actual table names (+ DS name in parens) instead of table IDs

## Risk

Not much. Option is only available to dust workspace atm

### CODE REVIEW

File: front/lib/api/assistant_builder/server_side_helpers.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  isTablesQueryConfiguration,
+} from "@dust-tt/types";
+
+import type { AssistantBuilderInitialState } from "@app/lib/api/assistant_builder/types";
+import { getDataSource } from "@app/lib/api/data_sources";
+import type { Authenticator } from "@app/lib/auth";
+import { tableKey } from "@app/lib/client/tables_query";
+import logger from "@app/logger/logger";
+
+export async function buildInitialState({
```
Comment (fontanierh) [inline front/lib/api/assistant_builder/server_side_helpers.ts:21]: had to move that stuff here for nextJS bundling reasons -- it kept trying to import it into the client, and some of the imports here don't work in the browser.

File: front/lib/api/assistant_builder/types.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import type {
```
Comment (fontanierh) [inline front/lib/api/assistant_builder/types.ts:1]: Had to move all that stuff here, as the types are needed in a server-side only file

**Review by spolu — CHANGES_REQUESTED** [review]
As per IRL let's keep all component related code in their directory :+1:

**Review by spolu — APPROVED** [review]
LGTM

Feel free to merge once `dataSourceByName` is used instead of reloading

**Pull request merged**